scripts/Plotter.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: Intel Corporation
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import json
+import os
+import re
+from dataclasses import dataclass
+
+import click
+import matplotlib.pyplot as plt
+
+
+@dataclass
+class BenchResults:
+    copy: float
+    scale: float
+    add: float
+    triad: float
+
+
+class Plotter:
+    def __init__(self) -> None:
+        self.plots_path_shp = "."
+        self.plots_path_mhp = "."
+        self.bench_path_shp = "."
+        self.bench_path_mhp = "."
+        self.create_directories()
+
+    def create_directories(self):
+        if not os.path.exists(self.plots_path_shp):
+            os.makedirs(self.plots_path_shp)
+        if not os.path.exists(self.plots_path_mhp):
+            os.makedirs(self.plots_path_mhp)
+        if not os.path.exists(self.bench_path_shp):
+            os.makedirs(self.bench_path_shp)
+        if not os.path.exists(self.bench_path_mhp):
+            os.makedirs(self.bench_path_mhp)
+
+    def clean_directories(self):
+        if not os.path.exists(self.plots_path_shp):
+            for file in os.listdir(self.plots_path_shp):
+                os.remove(os.path.join(self.plots_path_shp), file)
+        if not os.path.exists(self.plots_path_mhp):
+            for file in os.listdir(self.plots_path_mhp):
+                os.remove(os.path.join(self.plots_path_mhp), file)
+        if not os.path.exists(self.bench_path_shp):
+            for file in os.listdir(self.bench_path_shp):
+                os.remove(os.path.join(self.bench_path_shp), file)
+        if not os.path.exists(self.bench_path_mhp):
+            for file in os.listdir(self.bench_path_mhp):
+                os.remove(os.path.join(self.bench_path_mhp), file)
+
+    @staticmethod
+    def _get_dev_num(filename: str):
```

Comment:
Why the number of devices is not stored in .json? All data necessary for plot should be there IMO. File names are auxiliary, eg. for human operators chosing data for manual plot.

Reply:
Because json is automatically generated by gbench and we do not control its content.

Reply:
See https://github.com/oneapi-src/distributed-ranges/blob/cb32defeff7893c10e360f7af50c41eec8c9a350/benchmarks/gbench/mhp/mhp-bench.cpp#L35 for adding more info to the json.

Reply:
number of devices added to json output files